main.cpp

```diff
@@ -237,13 +329,76 @@ struct RolandTB303
         void savingSequence(int stepSize, int numberOfStepsEnabled, int numberOfLoops, int tempo);
         bool savingToExternal(int stepCount, std::string fileType = "TB303.p");
         void displaySavingProgress(int timeRemaining, float savingRate);
+
+        // new function
+        void recallPattern(char X);
     };
 
     void programSequence(RolandTB303 sequence);
     void filterSignal(float cutoffFrequency);
     float tempoAdjust(RolandTB303 tempo);
+
+    // new function
+    int syncMidi(int midiNotesIn);
 };
 
+void RolandTB303::SavePattern::recallPattern(char X)
+{    
+    if (X == 'A')
+    {
+    std::cout << "Loading previous sequence... " << X << "\n";
+    int sequenceMemory = 25;
+    std::cout << "Sequence step total: " << sequenceMemory << "\n";
+        
+    for (int seq = 0; seq < 25; ++seq)
+    {
+        std::cout << "Step " << (seq + 1) << " loaded\n";
+    }
+    }
```

Comment:
Fix this indentation
- [ ] revise

Reply:
Thanks, I do attempt to fully-proof read before sending. I did find another and fixed that too.